src/pricer.py

```diff
@@ -380,6 +350,9 @@ def __init__(self, contract: AmericanContract, model: MarketModel, params: TreeP
             tree_method = SimpleBinomialTree(self.contract, self.model, self.params)
         self.tree_method = tree_method
 
+    def pre_final_value(self, spot: dict[float, float], step: int, discounted_continuation_value: float) -> float:
+        raise RuntimeError('pre_final_value must be overriden in child classes.')
```

Comment:
Remove the error, inherit the class from ABC and mark the this method as abstractmethod.

Reply:
did it, unit tests run, does it look good?